blueocean-rest/src/main/java/io/jenkins/blueocean/rest/pageable/PagedResponse.java

```diff
@@ -41,34 +42,33 @@ public Object invoke(StaplerRequest request, StaplerResponse response, Object in
             }
             final Pageable<?> resp = (Pageable<?>) target.invoke(request, response, instance, arguments);
 
-            return new HttpResponse() {
-                @Override
-                public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object node) throws IOException, ServletException {
-                    int start = (req.getParameter("start") != null) ? Integer.parseInt(req.getParameter("start")) : 0;
-                    int limit = (req.getParameter("limit") != null) ? Integer.parseInt(req.getParameter("limit")) : DEFAULT_LIMIT;
+            return (HttpResponse) ( req, rsp, node ) -> {
+                int start = (req.getParameter("start") != null) ? Integer.parseInt(req.getParameter("start")) : 0;
+                int limit = (req.getParameter("limit") != null) ? Integer.parseInt(req.getParameter("limit")) : DEFAULT_LIMIT;
 
-                    if(start < 0){
-                        start = 0;
-                    }
+                if(start < 0){
+                    start = 0;
+                }
 
-                    if(limit < 0){
-                        limit = DEFAULT_LIMIT;
-                    }
-                    Object[] page = Iterators.toArray(resp.iterator(start, limit), Object.class);
-                    String url = req.getOriginalRequestURI();
-
-                    String separator = "?";
-                    if(req.getQueryString() != null){
-                        String q = getQueryString(req.getQueryString(), "start", "limit");
-                        if(q.length()>0){
-                            url += "?"+q;
-                            separator = "&";
-                        }
-                    }
-                    rsp.setHeader("Link", "<" + url + separator + "start=" + (start + limit) + "&limit="+limit + ">; rel=\"next\"");
+                if(limit < 0){
+                    limit = DEFAULT_LIMIT;
+                }
+
+                Object[] page = StreamSupport.stream(Spliterators.spliteratorUnknownSize(resp.iterator( start, limit), Spliterator.SORTED), false)
```

Comment:
How about some comments for what is being done here?  

Also formatting:
```suggestion
                Object[] page = StreamSupport.stream(Spliterators.spliteratorUnknownSize(resp.iterator(start, limit), Spliterator.SORTED), false)
```

Reply:
you mean more than it was documented before :P

Reply:
🤷

Reply:
At least github comment?  Have pity on your reviewers.